File: Implementation/2828.cpp
#include <iostream>
using namespace std;

int main() {
	int n, m, j;
	cin >> n >> m >> j;
	int left = 1, right = left + m - 1;
	int answer = 0;
	for(int i = 0; i < j; i++){
		int fall;
		cin >> fall;
		
		if(fall >= left && fall <= right) continue;
		
		if(right < fall){
			int offset = fall - right;
			left += offset;
			right += offset;
			answer += offset;
		} else if(left > fall){
			int offset = left - fall;
			left -= offset;
			right -= offset;
			answer += offset;
		}
	}
	cout << answer;
	return 0;
}

File: Implementation/2828.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin', 'utf8').trim().split('\n');
const [n, m] = input[0].split(' ').map(Number);
const j = parseInt(input[1]);

let left = 1;
let right = m;
let answer = 0;

for(let i = 2; i < 2 + j; i++){
    const fall = parseInt(input[i]);
	if(fall >= left && fall <= right) continue;
	
	if(fall > right){
		const offset = fall - right;
		left += offset;
		right += offset;
		answer += offset;
	} else if(fall < left){
		const offset = left - fall;
		left -= offset;
		right -= offset;
		answer += offset;
	}
}
console.log(answer);
